src/services/ActivityService.ts

```diff
@@ -84,39 +84,37 @@ export class ActivityService {
 
     /**
      * Ends an activity with the given id
+     * if the visit was already closed, the response will be 200 and the flag visitClosed will be set to true
      * @param id - id of the activity to end
-     * @returns Promise void
+     * @returns Promise<{visitClosed: boolean}>
      */
-    public async endActivity(id: string): Promise<void> {
-        return this.dbClient.get({ id })
-            .then(async (result: DocumentClient.GetItemOutput): Promise<void> => {
-                // Result checks
-                if (result.Item === undefined) {
-                    throw new HTTPResponse(404, { error: HTTPRESPONSE.NOT_EXIST });
-                }
-
-                if (result.Item.endTime !== null) {
-                    throw new HTTPResponse(403, { error: HTTPRESPONSE.ALREADY_ENDED });
-                }
-
-                const activity: IActivity = result.Item as IActivity;
-
-                // Assign the endTime
-                const endTime: string = new Date().toISOString();
-                Object.assign(activity, { endTime });
+    public async endActivity(id: string): Promise<{ visitClosed: boolean }> {
+        try {
+            const result: DocumentClient.GetItemOutput = await this.dbClient.get({ id });
+            if (result.Item === undefined) {
+                throw new HTTPResponse(404, { error: HTTPRESPONSE.NOT_EXIST });
+            }
 
-                await this.dbClient.put(activity);
-            })
-            .catch((error: AWSError | HTTPResponse) => {
-                // If we get HTTPResponse, we rethrow it
-                if (error instanceof HTTPResponse) {
-                    console.error(`Error occurred: ${error.body}`);
-                    throw error;
-                }
+            if (result.Item.endTime !== null) {
+                console.log(`Visit with ID ${id} is already closed`);
+                return { visitClosed: true };
+            }
+            
+            const activity: IActivity = result.Item as IActivity;
+            activity.endTime = new Date().toISOString();
+            await this.dbClient.put(activity);
+
+            return { visitClosed: false };
+        } catch (error) {
+            // If we get HTTPResponse, we rethrow it
+            if (error instanceof HTTPResponse) {
+                console.error(`Error occurred: ${error.body}`);
```

Comment:
Why not log the error regardless, would seem to be valuable regardless of whether a HTTP problem, Dynamo, or something else.

Reply:
removed the catch block since we are throwing a response in the try block and refactored to log the message once the visitId is not found